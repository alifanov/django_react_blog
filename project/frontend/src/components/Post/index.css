.post-tags {
  font-size: 13px;
  margin-bottom: 10px;
}
