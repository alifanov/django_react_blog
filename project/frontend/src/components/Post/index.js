import React from "react";
import PropTypes from "prop-types";

import { Card, Button, CardBody, CardText, CardHeader } from "reactstrap";
import { FaTimesCircle } from "react-icons/fa";
import Moment from "react-moment";

import "./index.css";

const Post = props => {
  return (
    <Card className="mb-4 post-item">
      <CardHeader className={"d-flex justify-content-between"}>
        <div className={"d-flex align-items-center"}>{props.title}</div>
        <FaTimesCircle
          onClick={() => props.onDelete(props.id)}
          style={{ cursor: "pointer" }}
        />
      </CardHeader>
      <CardBody>
        <CardText>{props.text}</CardText>
        <div className="post-tags">
          {props.tags.map(t => (
            <span key={t}>#{t}</span>
          ))}
        </div>
        <div className="d-flex justify-content-between align-items-center">
          <div className="btn-group">
            <Button size={"sm"} outline color={"secondary"}>
              Read
            </Button>
          </div>
          <small className="text-muted">
            Published:&nbsp;
            <Moment format={"DD.MM.YYYY HH:mm"}>{props.created_at}</Moment>
          </small>
        </div>
      </CardBody>
    </Card>
  );
};

Post.propTypes = {
  onDelete: PropTypes.func
};

export default Post;
