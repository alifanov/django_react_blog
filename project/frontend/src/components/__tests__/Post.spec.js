import React from "react";
import Post from "../Post";
import Moment from "react-moment";

import { shallow, mount } from "../../enzyme";
import { MemoryRouter } from "react-router-dom";

import renderer from "react-test-renderer";

describe(">>> Post test", () => {
  let post;
  const content = { title: "Title1", text: "Text1", tags: [] };
  beforeEach(() => {
    post = shallow(<Post {...content} />);
  });

  it("+++ render Post component", () => {
    expect(post.length).toEqual(1);
  });
  it("+++ check text for Post", () => {
    expect(
      post
        .find("CardHeader>div")
        .at(0)
        .props().children
    ).toEqual("Title1");
  });
  it("+++ check for delete Post", () => {
    const mockDeleteFn = jest.fn();
    const post = shallow(<Post onDelete={mockDeleteFn} {...content} />);
    const component = post.dive();
    component
      .find("CardHeader>FaTimesCircle")
      .at(0)
      .simulate("click");

    expect(mockDeleteFn.mock.calls.length).toEqual(1);
  });
});

describe(">>>Post --- Snapshot", () => {
  it("+++capturing Snapshot of Post", () => {
    const content = {
      title: "Title1",
      text: "Text1",
      tags: [],
      created_at: new Date(2018, 1, 1, 11, 11, 11, 11)
    };
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <Post {...content} />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
