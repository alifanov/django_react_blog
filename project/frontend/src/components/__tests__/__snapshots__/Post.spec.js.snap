// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Post --- Snapshot +++capturing Snapshot of Post 1`] = `
<div
  className="mb-4 post-item card"
>
  <div
    className="d-flex justify-content-between card-header"
  >
    <div
      className="d-flex align-items-center"
    >
      Title1
    </div>
    <svg
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "cursor": "pointer",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
      />
    </svg>
  </div>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      Text1
    </p>
    <div
      className="d-flex justify-content-between align-items-center"
    >
      <div
        className="btn-group"
      >
        <button
          aria-label={null}
          className="btn btn-outline-secondary btn-sm"
          onClick={[Function]}
        >
          Read
        </button>
      </div>
      <small
        className="text-muted"
      >
        Published:Â 
        <time
          dateTime={"2018-02-01T08:11:11.011Z"}
        >
          01.02.2018 11:11
        </time>
      </small>
    </div>
  </div>
</div>
`;
